vowpalwabbit/exploration.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include <stdint.h>
+#include <stdexcept>
+#include <algorithm>
+#include <numeric>
+#include <cstring>
+
+namespace exploration
+{
+  template<typename OutputIt>
+  void epsilon_greedy(float epsilon, uint32_t top_action, OutputIt pdf_first, OutputIt pdf_last)

```

Comment:
I'm still not fully clear on the portability of this interface.

Reply:
not sure what you mean by portability. Exposing it to other languages through bindings? If yes, neither this or the previous version fit. Based on the path one chooses (e.g. using swig bound vectors in python or plain c-style pointers) you'll have to write slim adapters in C/C++. Those adapters will have an easily live calling this function w/o memory copy.